source/common/io/io_uring_worker_impl.cc

```diff
@@ -0,0 +1,240 @@
+#include "source/common/io/io_uring_worker_impl.h"
+
+#include <sys/socket.h>
+
+#include "io_uring_impl.h"
+
+namespace Envoy {
+namespace Io {
+
+IoUringSocketEntry::IoUringSocketEntry(os_fd_t fd, IoUringWorkerImpl& parent)
+    : fd_(fd), parent_(parent) {}
+
+void IoUringSocketEntry::unlink() {
```

Comment:
nit: consider just inlining this if it's only going to be used in the one place below.